### PR TITLE
[10.x] Set CallQueuedHandler queue-related properties from queued EventListener

### PR DESCRIPTION
**Pull request by cosmastech**
CallQueuedHandler does not respect a queued Event Listener's properties, they just default to whatever is set on CallQueuedHandler.

This PR will carry over those properties.

This resolves at least one of the issues identified in https://github.com/laravel/framework/issues/46756

I'll add test cases if there is any reason to.

### CODE REVIEW

File: src/Illuminate/Events/CallQueuedListener.php [hunk 1]
```diff
@@ -123,12 +124,43 @@ protected function setJobInstanceIfNecessary(Job $job, $instance)
...
+        return $handler;
+    }
+
     /**
      * Call the failed method on the job instance.
      *
      * The event instance and the exception will be passed.
      *
-     * @param  \Throwable  $e
+     * @param  Throwable  $e
```
Comment (driesvints) [inline src/Illuminate/Events/CallQueuedListener.php:163]: Please revert this change.

**Comment by taylorotwell** [comment]
We already do this here: https://github.com/laravel/framework/blob/10.x/src/Illuminate/Events/Dispatcher.php#L618

???
